src/client/automation/playback/click/index.js

```diff
@@ -225,12 +225,20 @@ export default class ClickAutomation {
         return cursor
             .buttonUp()
             .then(() => {
-                var prevElementForEvent = this.eventArgs.element;
-
-                this.eventArgs            = this._calculateEventArguments();
-                this.eventState.skipClick = prevElementForEvent !== this.eventArgs.element;
-
-                eventSimulator.mouseup(this.eventArgs.element, this.eventArgs.options);
+                var topElementEventArgs = this._calculateEventArguments();
```

Comment:
I agree, that this.eventArgs.options and this.eventArgs.point should be recalculated, but click should be raised over previous element.